server/src/main/java/org/elasticsearch/transport/TransportSettings.java

```diff
@@ -132,6 +132,12 @@
             Arrays.asList("internal:coordination/fault_detection/*"),
             Function.identity(), Setting.Property.Dynamic, Setting.Property.NodeScope);
 
+    // Time that processing an inbound message on a transport thread may take at the most before a warning is logged
+    public static final Setting<TimeValue> SLOW_OPERATION_THRESHOLD_SETTING =
+            Setting.positiveTimeSetting("transport.slow_operation_logging_threshold", TimeValue.timeValueMillis(300),
```

Comment:
`300ms` since our default resolution on the timer thread is `200ms`. This is already plenty IMO (3 requests per second isn't great on a transport thread) but not so low that any cgroup throttling or other system slowness instantly results in massive log spam.

Reply:
I'm still a bit concerned about log spam -- we aren't aware of anything that blocks the transport threads for so long today but that might just be because we don't surface it yet, and some configs might be hitting this a lot. `#processors` threads times 3 messages per second is too much for my taste. WDYT about a simple rate limit to avoid logging more than one of these per ten seconds or something? That'd be enough to point us in the right direction.

Reply:
I think it might be kind of nice to be able to see bursts here (thinking about spotting something like CPU throttling)? (if you have a time period where there's a bunch of logging across multiple threads for all kinds of messages without a clear pattern or so). I guess we could use something like the log4j burst filter here to capture that kind of thing and still rate limit but what this:
We could just do something like a hard `5s` timeout above which we always `WARN` and make this a `DEBUG` at `300ms`?

Reply:
I'd be happy with `5s` -- I think this would surface problems bad enough to drop nodes from the cluster without risking too much junk.

No need for a separate debug logging threshold IMO -- given that it would need user involvement to see more events in the logs, they may as well just reduce `transport.slow_operation_logging_threshold`.

Reply:
Fair point -> pushed the change to 5s :)